indexes/plugin.js

```diff
@@ -98,15 +98,18 @@ module.exports = class Plugin {
         processedSeq = status.processed
         if (this.onLoaded) {
           this.onLoaded(() => {
+            processedOffset = status.offset
```

Comment:
What?? This wasn't there before? OMG how could I have not seen that before, I looked at this file so many times and at `processedOffset` so many times. :see_no_evil:

Reply:
By the way, I think this code should be changed so that we first set `processedOffset` and `this.offset` **and then** we call `this.onLoaded()`. That's because `this.onLoaded()` could do some inversion of control. It could trigger other computation to happen which *could* trigger `onRecord()` to be called, and by the time `onRecord` is called we MUST have `processedOffset` set correctly.

Seems to me like as soon as we load META from leveldb, there is no reason to delay the setting of `processedOffset`, we should just write it ASAP.

But still, call `this._stateLoaded.resolve()` only after `this.onLoaded()` is done, to preserve current code behavior.

Reply:
makes sense, good catch